core/src/main/java/org/apache/calcite/rel/rules/SortRemoveConstantKeysRule.java

```diff
@@ -77,8 +78,13 @@ protected SortRemoveConstantKeysRule(Config config) {
       return;
     }
 
+    final RelCollation collation =
+        RelCollationTraitDef.INSTANCE.canonize(RelCollations.of(collationsList));
     final Sort result =
-        sort.copy(sort.getTraitSet(), input, RelCollations.of(collationsList));
+        sort.copy(
```

Comment:
Can we add a test in RelOptRulesTest.xml to illustrate this?

Reply:
`RelOptRulesTest#testSortRemovalOneKeyConstant` with the proposed adjustment tests this (otherwise the adjusted test fails). I have described it a bit more in detail on the Jira ticket.